templates/events/events.ical

```diff
@@ -0,0 +1,68 @@
+{# THIS FILE MUST BE SAVED WITH LF ENDINGS ONLY, NO CRLF FOR YOU -#}
+
+{%- import "macros/formatters.html" as formatters -%}
+{% import "events/macros/event_format.html" as event_format -%}
+
+BEGIN:VCALENDAR
+VERSION:2.0
+PRODID:-//uwcs.co.uk//Events//EN
+CALSCALE:GREGORIAN
+METHOD:PUBLISH
+X-WR-CALDESC:Events hosted by the University of Warwick Computing Society
+X-WR-CALNAME:UWCS Events
+X-WR-TIMEZONE:GMT
+
+{%- set events = get_section(path="events/_index.md") -%}
+{% for t in events.subsections -%}
+  {% if t is containing("archive") -%}{%- continue -%}{%- endif -%}
+  {% set term = get_section(path=t) -%}
+  {% for week in term.subsections | sort -%}
+    {% if week is containing("repeat") -%}{%- continue -%}{%- endif -%}
+    {% set w = get_section(path=week) -%}
+    {% set week_base = w.extra.base_date -%}
+    
+    {% set rep_path = w.components[0] ~ "/" ~ w.components[1] ~ "/repeat/_index.md" -%}
+    {% if term.subsections is containing(rep_path) -%}
+      {# Get repeating events -#}
+      {% set repeating = get_section(path=rep_path) -%}
+      {% set all_rep_events = repeating.pages | sort(attribute="date") -%}
+
+      {# Filter events by the current week number -#}
+      {% set week_no = w.components | last -%}
+      {% set_global rep_events = [] -%}
+      {% for e in all_rep_events -%}
+        {% if not e.extra.weeks or e.extra.weeks is containing(week_no) -%}
+          {% set_global rep_events = rep_events | concat(with=e) -%}
+        {% endif -%}
+      {% endfor -%}
+    {% else -%}
+      {% set_global rep_events = [] -%}
+    {% endif -%}
+
+    {% set events = w.pages | concat(with=rep_events) -%}
+
+    {% for page in events -%}
+
+{% set repeat = page.permalink is containing("/repeat/") %}
```

Comment:
identation--

Reply:
the ical has to be flat, but the rest can still be indented (only other compromise is to only indent the code lines and not the ical lines which is far more cursed